src/app/components/NetworkBadge/NetworkBadge.tsx

```diff
@@ -1,21 +1,28 @@
 import Image from "next/image";
 
 import testnetIcon from "./testnet-icon.png";
+import { Network } from "@/utils/wallet/wallet_provider";
 
 // This component can also be used rendering based on the network type
 interface NetworkBadgeProps {}
 
 export const NetworkBadge: React.FC<NetworkBadgeProps> = () => {
   return (
     <div className="absolute left-2 top-[6rem]">
-      <Image src={testnetIcon} alt="Testnet" className="w-[10rem]" />
-      {/*
-      currently the text is absolutely positioned
-      since the image has a shadow
-       */}
-      <p className="absolute left-4 top-[4rem] text-sm dark:text-neutral-content">
-        Testnet
-      </p>
+      {
+        process.env.NEXT_PUBLIC_NETWORK === Network.TESTNET ||
+        process.env.NEXT_PUBLIC_NETWORK === Network.SIGNET &&
```

Comment:
```suggestion
        (process.env.NEXT_PUBLIC_NETWORK === Network.TESTNET ||
        process.env.NEXT_PUBLIC_NETWORK === Network.SIGNET) &&
```

`&&` takes precedence over `||`

Reply:
Also, we should include regtest here